src/main/kotlin/com/autonomousapps/model/declaration/Configurations.kt

```diff
@@ -77,7 +82,11 @@ internal object Configurations {
 
   // we want dependency buckets only
   fun Configuration.isForRegularDependency() =
-    !isCanBeConsumed && !isCanBeResolved && isForRegularDependency(name)
+    // do not check '!isCanBeConsumed && !isCanBeResolved' due to https://github.com/gradle/gradle/issues/20547 or
+    // similar situations. Other plugins or users (although not recommended) might change these flags. Since we know
+    // the exact names of the Configurations we support (based on to which source set they are linked) this check
+    // is not necessary.
+    isForRegularDependency(name)
```

Comment:
I hope your are comfortable with this change.

Reply:
The comment + a comprehensive test suite gives me great confidence :)